### DISCUSSION TITLE
hide body param from network tab

### DISCUSSION THREAD

**Discussion by breakingbad45** [pocketbase/pocketbase#3622] [Q&A]
can we hide the body param from the browser network>Request tab?

**Comment by ganigeorgiev** [comment]
You can't hide the request payload that the client is sending because they are the one who initiated the request.

I guess you can create a proxy with a custom route with different parameters than the original ones but I can't see the benefit of doing this as users can just send request to the custom route.

Whatever is downloaded and executed on the client-side, the user has full control over it, even if you obfuscate your JS code.

You've asked several security related questions but I'm failing to understand what is the end goal as you never bothered to provide details about your use case. Could you elaborate or at least explain your concerns? Maybe there is a better solution to what you are trying to do.

**Reply by breakingbad45** [reply]
I am facing an issue where someone is using my API to insert data into my server’s database without authorization. To enhance security, I want to restrict access to only registered users and approve requests only from my domain

**Reply by ganigeorgiev** [reply]
> and approve requests only from my domain

This doesn't really make sense. If the request is sent from the client, then the client (aka. the user device) is the one who sent the request, not your domain (aka. your server).

Use the Collection API rules to limit the access and the operations for your data - https://pocketbase.io/docs/api-rules-and-filters/.
